### PR TITLE
Add ability to stream files directly from the storage backend instead of

### PR DESCRIPTION
**Pull request by asgoel**
returning a redirect URL

### CODE REVIEW

File: pypicloud/__init__.py [hunk 1]
```diff
@@ -143,6 +143,16 @@ def run_postfork_hooks():
     config.registry.fallback = fallback_mode
     config.registry.always_show_upstream = always_show_upstream
 
+    stream_files = settings.get("pypi.stream_files")
+    if stream_files is None:
+      stream_files = False
+    else:
+      stream_files = asbool(stream_files)
+    config.registry.stream_files = stream_files
```
Comment (stevearc) [inline pypicloud/__init__.py]: A more concise option:

```suggestion
    config.registry.stream_files = asbool(settings.get("pypi.stream_files", False))
```

File: pypicloud/views/api.py [hunk 2]
```diff
@@ -106,9 +106,21 @@ def download_package(context, request):
         package, data = fetch_dist(request, dist.name, source_url)
         disp = CONTENT_DISPOSITION.tuples(filename=package.filename)
         request.response.headers.update(disp)
+        if request.registry.max_age:
```
Comment (stevearc) [inline pypicloud/views/api.py]: I think it's safe to include the header even if `max_age = 0` (as you're doing below)

File: pypicloud/__init__.py [hunk 3]
```diff
@@ -143,6 +143,16 @@ def run_postfork_hooks():
...
+
+    max_age = int(settings.get("pypi.package_max_age", 0))
+    config.registry.max_age = max_age
```
Comment (stevearc) [inline pypicloud/__init__.py]: Oh forgot: small request to change this to `package_max_age` to make it clearer that it's the max age specifically for packages.

**Comment by asgoel** [comment]
@stevearc mind taking another look? just force pushed, and added tests as well as the documentation update.

**Comment by stevearc** [comment]
Awesome, thanks!

Merged into [the 1.0 branch](https://github.com/stevearc/pypicloud/commits/1.0). Will go out in the next release

**Comment by asgoel** [comment]
Hey @stevearc, did you ever end up releasing this?

**Comment by stevearc** [comment]
Not yet. I usually wait for other PRs to come in and then bundle them in a release, but the last couple months have been pretty quiet. I'll do it this weekend

**Pull request merged**